feature/authentication-logic

Mohamed-Elhanafy opened the pull request:
implement authentication local datasource , remote datasource and domain .

Mohamed-Elhanafy committed:
Add remote authentication data source with Retrofit integration

- Implement AuthenticationApi, AuthenticationRemoteDataSource, and its implementation
- Add exception handling for authentication and server errors
- Configure new Gradle module for remote authentication data source
- Update dependencies and module references for authentication feature
- Extend NetworkModule with Retrofit and OkHttp setup for authentication
- Add ProGuard and consumer rules for new module

Mohamed-Elhanafy committed:
Add local authentication data source with SharedPreferences implementation

Mohamed-Elhanafy committed:
Add authentication repository implementation and integrate with DI module

Mohamed-Elhanafy committed:
Add domain authentication module with repository interface and login use case integration

Mohamed-Elhanafy committed:
Merge branch 'develop' into feature/authentication-logic

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/java/com/paris_2/aflami/di/DataSourceModule.kt

Joseph-Sameh-0 commented on app/build.gradle.kts:
```diff
@@ -177,18 +179,30 @@ dependencies {
+    implementation(project(Modules.DATASOURCE_REMOTE_AUTHENTICATION))
+
+    implementation(project(Modules.DATASOURCE_LOCAL_MOVIE))
+    implementation(project(Modules.DATASOURCE_LOCAL_MOVIE))
```
Don't duplicate the dependency

Joseph-Sameh-0 commented on app/src/main/java/com/paris_2/aflami/di/DataSourceModule.kt:
```diff
@@ -53,4 +57,7 @@ val dataSourceModule = module {
     singleOf(::TvShowSimilarLocalDataSourceImpl) bind TvShowSimilarLocalDataSource::class
     singleOf(::GenresInteractionDataSourceImpl) bind GenresInteractionDataSource::class
     singleOf(::MovieSimilarLocalDataSourceImp) bind MovieSimilarLocalDataSource::class
+    singleOf(::GenresInteractionDataSourceImpl) bind GenresInteractionDataSource::class
```
don't duplicate

Mohamed-Elhanafy committed:
Add unit tests for authentication repository, remote data source, and login use case

Joseph-Sameh-0 commented on datasource/remote/authentication/build.gradle.kts:
```diff
@@ -0,0 +1,66 @@
+            verify {
+                rule {
+                    bound {
+                        minValue = 0
```
use the coverageMinValue not 0

Joseph-Sameh-0 reviewed (commented):
Update the release info

Mohamed-Elhanafy committed:
Add guest login flow to authentication domain, repository, and data sources

- Implement GuestSessionDto model and API endpoint for guest session creation
- Extend AuthenticationRemoteDataSource and AuthenticationRepository with guest login methods
- Add GuestLoginUseCase and integrate with DI module
- Update repository and data source implementations for guest login support
- Add unit tests for guest login use case, repository, remote and local data sources
- Refactor LoginUseCase to remove saveSessionId method
- Clean up build scripts and DI modules
- Integrate guest login invocation in MainActivity for testing

Mohamed-Elhanafy committed:
Remove guest login use case invocation and related dependencies from MainActivity

Joseph-Sameh-0 committed:
fix: Correct path for REPOSITORY_AUTHENTICATION module

The path for the `REPOSITORY_AUTHENTICATION` module in `Modules.kt` was corrected by removing an extra colon. It changed from `::repository:authentication` to `:repository:authentication`.

Joseph-Sameh-0 committed:
Refactor: Enhance error handling in login flow

The `login` method in `AuthenticationRepositoryImpl` now checks for a successful validation response from `remoteDataSource.validateWithLogin`. If the validation is not successful or the success field is null, the method returns `false`, preventing further execution of the login process.

Mohamed-Elhanafy committed:
Update release info for authentication domain, data source, and AuthInterceptor additions
